backend/src/routes/index.ts

```diff
@@ -0,0 +1,18 @@
+import { getLines } from './../controllers/lineController';
+import { getRooms, addNewRoom, deleteRoom, roomExist, getUserRooms } from './../controllers/roomController';
+import { register, login, getUsername } from './../controllers/userController';
+import express, { Router } from "express";
+import { authMiddleware } from '../middlewares/authMiddleware';
+import { validParams } from '../middlewares/paramsValidation';
+
+export const router: Router = express.Router();
+
```

Comment:
You get the REST names wrong. You should use plural naming like `/rooms/:name`, `/lines/:roomName` ect. read about rest path naming.